### PR TITLE
Allow Fauxton docroot to be overridden (#3333)

### PR DESCRIPTION
**Pull request by sklassen**
## Overview

This relates to issue #3333

If erlang's current working directory become unreadable, the fauxton (and any other relative paths) yields 404 Not found. 

## Testing recommendations

I changed the couchdb.in and re-ran the server with an absolute path.

```
simon@sigyn:couchdb$ export COUCHDB_FAUXTON_DOCROOT=/home/simon/src/github.com/apache/couchdb/share/www/
simon@sigyn:couchdb$ ./dev/run -n 1
```

```
simon@sigyn:couchdb$ ./dev/remsh 
Erlang/OTP 22 [erts-10.6.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

Eshell V10.6.4  (abort with ^G)
(node1@127.0.0.1)1> pwd().
/home/simon/src/github.com/apache/couchdb
ok
(node1@127.0.0.1)2> cd("/tmp").
/tmp
ok
```
Fauxton website return 200 and was operational. 

## Related Issues or Pull Requests

None.

## Checklist

- [X] Code is written and works correctly
- [X] Changes are covered by tests
- [X] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [X] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

**Comment by janl** [comment]
That PR is not quite right :)

**Comment by sklassen** [comment]
> That PR is not quite right :)

Let me try again. I thought I had sync'd with 3.x branch.
